File: src/main/java/compiler/block/comments/CommentBlock.java
package compiler.block.comments;

import compiler.block.Block;

public class CommentBlock extends Block {

    private String type = "comment";


    public CommentBlock(Block superBlock) {
        super(superBlock, false, false);

    }

    @Override
    public void init() {

    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return null;
    }

    @Override
    public String getBodyCode() {
        return null;
    }

    @Override
    public String getClosingCode() {
        return null;
    }

    @Override
    public String toString() {
        return "Comment";
    }
}

File: src/main/scala/compiler/block/comments/CommentBlock.scala
package compiler.block.comments

import compiler.block.Block

/**
  * Represents a code comment
  * E.g. "// This is a comment"
  *
  * @param superBlockInit
  */
class CommentBlock(var superBlockInit: Block) extends Block(superBlockInit, false, false) {
  private val _type: String = "comment"

  // Called after file is parsed
  override def init() {}

  /* Symbol table information */
  override def getName: String = ""

  override def getValue: String = ""

  override def getType: String = _type

  /* Opening and closing byte code */
  def getOpeningCode: String = ""

  def getClosingCode: String = ""

  /* toString method */
  override def toString: String = "Comment"

}